packages/request-logic/src/request-logic.ts

```diff
@@ -325,61 +325,24 @@ export default class RequestLogic implements RequestLogicTypes.IRequestLogic {
   public async getRequestFromId(
     requestId: string,
   ): Promise<RequestLogicTypes.IReturnGetRequestFromId> {
-    const resultGetTx = await this.transactionManager.getTransactionsByChannelId(requestId);
-    const actions = resultGetTx.result.transactions
-      // filter the actions ignored by the previous layers
-      .filter(action => action !== null)
-      .sort((a: any, b: any) => a.timestamp - b.timestamp);
-
-    // tslint:disable-next-line:prefer-const
-    let { ignoredTransactions, keptTransactions } = this.removeOldPendingTransaction(actions);
-
-    // array of transaction without duplicates to avoid replay attack
-    const actionsTimestampedWithoutDuplicates = Utils.uniqueByProperty(
-      keptTransactions
-        .map((t: any) => {
-          try {
-            return {
-              action: JSON.parse(t.transaction.data),
-              state: t.state,
-              timestamp: t.timestamp,
-            };
-          } catch (e) {
-            // We ignore the transaction.data that cannot be parsed
-            ignoredTransactions.push({
-              reason: 'JSON parsing error',
-              transaction: t,
-            });
-            return;
-          }
-        })
-        .filter((elem: any) => elem !== undefined),
-      'action',
-    );
+    const {
+      ignoredTransactions,
+      requestStateConfirmed,
+      requestStatePending,
+      transactionManagerMeta,
+    } = await this.computeRequestFromRequestId(requestId);
 
-    // Keeps the transaction ignored
-    ignoredTransactions = ignoredTransactions.concat(
-      actionsTimestampedWithoutDuplicates.duplicates.map(tx => {
-        return {
-          reason: 'Duplicated transaction',
-          transaction: tx,
-        };
-      }),
+    const pending = this.computeDiffBetweenPendingAndConfirmedRequestState(
```

Comment:
`computeDiffBetweenPendingAndConfirmedRequestState`, once again I find this name ambiguous

Reply:
`diffRequestStates`?